subprojects/kotlin-dsl-tooling-builders/src/integTest/groovy/org/gradle/kotlin/dsl/tooling/builders/KotlinDslScriptsModelSpec.groovy

```diff
@@ -16,48 +16,46 @@
 
 package org.gradle.kotlin.dsl.tooling.builders
 
-import groovy.transform.stc.ClosureParams
-import groovy.transform.stc.SimpleType
 import org.gradle.integtests.fixtures.AbstractIntegrationSpec
 import org.gradle.integtests.fixtures.build.TestProjectInitiation
+import org.gradle.integtests.tooling.fixture.TestOutputStream
 import org.gradle.integtests.tooling.fixture.ToolingApi
-import org.gradle.tooling.GradleConnectionException
-import org.gradle.tooling.ProjectConnection
-import org.gradle.tooling.model.kotlin.dsl.KotlinDslScriptModel
+import org.gradle.tooling.model.kotlin.dsl.KotlinDslScriptsModel
 import org.junit.Rule
 import org.junit.rules.RuleChain
 
-class KotlinDslScriptsModelSpec extends AbstractIntegrationSpec implements KotlinScriptsModelFetcher, TestProjectInitiation {
-    def toolingApi = new ToolingApi(distribution, temporaryFolder)
+class KotlinDslScriptsModelSpec extends AbstractIntegrationSpec implements TestProjectInitiation{
+    @Delegate
+    ToolingApi toolingApi = new ToolingApi(distribution, temporaryFolder)
+    TestOutputStream stderr = new TestOutputStream()
+    TestOutputStream stdout = new TestOutputStream()
 
     @Rule
     public RuleChain cleanupRule = RuleChain.outerRule(temporaryFolder).around(toolingApi)
 
-    def <T> T withConnection(@DelegatesTo(ProjectConnection) @ClosureParams(value = SimpleType, options = ["org.gradle.tooling.ProjectConnection"]) Closure<T> cl) {
-        try {
-            toolingApi.withConnection(cl)
-        } catch (GradleConnectionException e) {
-            throw e
-        }
-    }
-
     def 'exceptions in different scripts are reported on the corresponding scripts'() {
 
         given:
-        toolingApi.requireIsolatedUserHome()
+        requireIsolatedUserHome()
 
         when:
         def spec = withMultiProject()
         spec.scripts["a"] << "throw RuntimeException(\"ex1\")"
         spec.scripts["b"] << "throw RuntimeException(\"ex2\")"
 
-        Map<File, KotlinDslScriptModel> singleRequestModels = kotlinDslScriptsModelFor(true).scriptModels
+
+        def model = loadToolingLeanModel(KotlinDslScriptsModel) {
+            KotlinScriptModulParameters.setModelParameters(it, true, true, [])
+        }
+
+        Map<File, KotlinDslScriptsModel> singleRequestModels = model.scriptModels
 
         then:
 
         singleRequestModels[spec.scripts["a"]].exceptions.size() == 1
         singleRequestModels[spec.scripts["b"]].exceptions.size() == 1
         singleRequestModels[spec.scripts["settings"]].exceptions.isEmpty()
-    }
 
+        validateOutput()
```

Comment:
:question: is it required to call `validateOutput()` in each test in order to fail on deprecation warnings? I would have expected this is automatic in all cross version tests and we add explicit expectations to make the tests pass.